src/CodeGeneration/OfferFriendlyOverloadsGenerator.cs

```diff
@@ -176,6 +184,15 @@ private static bool IsByteStar(TypeSyntax typeSyntax)
             return predefinedElementType?.Keyword.IsKind(SyntaxKind.ByteKeyword) ?? false;
         }
 
+        private static IEnumerable<ParameterSyntax> WhereIsIntPtrParameter(IEnumerable<ParameterSyntax> parameters, IEnumerable<ParameterSyntax> newParameters)
```

Comment:
This method seems super-highly specialized to your one purpose. Yet its name suggests it searches a sequence for `IntPtr` parameters, which is far simpler than what it actually does, and in fact does not fully wrap your logical operation which seems to be to test whether an colliding `IntPtr` overload has already been generated. 
Can you please move more of this method's caller's logic into this method and then give it a higher-level name (e.g. "IsIntPtrMethodGenerated")?